alerter.c

```diff
@@ -1,32 +1,41 @@
 #include <stdio.h>
 #include <assert.h>
+#include "alerter.h"
 
 int alertFailureCount = 0;
+enum environments{Test,Integration}environment;
 
-int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
-}
-
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
+    void AlertFarenheitToCelcius(float farenheit) 
+    {
+        celcius = (farenheit - 32) * 5 / 9;
+        printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+        if (environment == Test)
+        {
+            returncode = stub(celcius);
+        }
+        else
+        {
+            //Integration environment
+            returncode = network(celcius);
+        }
+        if (returncode != 200) 
+        {
         // non-ok response is not an error! Issues happen in life!
         // let us keep a count of failures to report
         // However, this code doesn't count failures!
         // Add a test below to catch this bug. Alter the stub above, if needed.
         alertFailureCount += 0;
+        }
     }
-}
 
 int main() {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
+    environment=Test;

```

Comment:
Production code should not have different paths for test and integration. The production code should behave in a definite way when given definite input , be it test environment or deployment. think of ways to integrate stubs during test and actual dependency in production environment.